proposals/collection-expressions.md

```diff
@@ -188,23 +187,26 @@ namespace System.Runtime.CompilerServices
     }
 }
 ```
-
-The attribute can be applied to `class`, `struct`, and `interface` types.
+The attribute can be applied to a `class`, `struct`, `ref struct`, or `interface`.
 The attribute is not inherited although the attribute can be applied to a base `class` or an `abstract class`.
 
-An error is reported for a collection literal if there are multiple `[CollectionBuilder]` attributes on the collection type.
+The collection type must have an [*iteration type*](https://github.com/dotnet/csharpstandard/blob/standard-v6/standard/statements.md#1295-the-foreach-statement).
 
-The *create method* must be a `static` method defined on the *builder type*.
-The *create method* must have a single parameter of a type `System.ReadOnlySpan<T0>` passed by value, and a return value of the *collection type* to which the attribute is applied.
+For the *create method*:
+- The *builder type* must be a non-generic `class` or `struct`.
+- The method must be defined on the *builder type* directly.
+- The method must be `public` and `static`.
+- The *arity* of the method must match the *arity* of the collection type.
+- The method must have a single parameter of type `System.ReadOnlySpan<E>`, passed by value, and there is an [*identity conversion*](https://github.com/dotnet/csharpstandard/blob/standard-v6/standard/conversions.md#1022-identity-conversion) between `E` and the [*iteration type*](https://github.com/dotnet/csharpstandard/blob/standard-v6/standard/statements.md#1295-the-foreach-statement) of the *collection type*.
+- There is an [*identity conversion*](https://github.com/dotnet/csharpstandard/blob/standard-v6/standard/conversions.md#1022-identity-conversion) between the method return type and the *collection type*.
 
-The *builder type* cannot be generic.
-The *builder method* can be generic and the *arity* of the builder method must match the *arity* of the *collection type*.
+An error is reported if the `[CollectionBuilder]` attribute does not refer to an invocable method with the expected signature.
 
-Overloads with distinct signatures on the builder type, and overloads on base types, are ignored.
+Method overloads on the *builder type* with distinct signatures are ignored. Methods declared on base types or interfaces are ignored.
 
-An error is reported for a collection literal if the `[CollectionBuilder]` attribute on the collection type does not represent an invocable method with the expected signature defined on the *builder type*.
+For a *collection expression* with a target type <code>C&lt;S<sub>0</sub>, S<sub>1</sub>, &mldr;&gt;</code> where the *type declaration* <code>C&lt;T<sub>0</sub>, T<sub>1</sub>, &mldr;&gt;</code> has an associated *builder method* <code>B.M&lt;U<sub>0</sub>, U<sub>1</sub>, &mldr;&gt;()</code>, the *generic type arguments* from the target type are applied in order &mdash; and from outermost containing type to innermost &mdash; to the *builder method*.
```

Comment:
I think this language is fine, but thought it might be useful to also link how the interceptors design described a similar thing. https://github.com/dotnet/roslyn/blob/main/docs/features/interceptors.md#arity